core/src/main/java/org/infinispan/remoting/rpc/RpcManager.java

```diff
@@ -21,186 +22,17 @@
  * @since 4.0
  */
 public interface RpcManager {
```

Comment:
AFAIK it would only save one singleton list and one singleton map, so I wouldn't say it's high priority. Remember, this PR isn't really about introducing new API, I just had to add `RpcManager.invokeRemotelyAsync()` because `RpcManager.invokeRemotelyWithFuture()` was so weird.